java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -243,6 +275,10 @@ public GeneratorResult generateJava() {
                         + ")"));
           }
         }
+
+        if (k.matches(".*-deprecated(\\d$)?$")) {
```

Comment:
Can we avoid this regexp continuously being computed on the hot path?
E.g. you can set a boolean here: https://github.com/fabric8io/kubernetes-client/pull/5007/files#diff-c55bf304c5b566fc02eed93f3960ca65807f07dcaa34dc364a43f953463df066R107

Reply:
`generateJava()` uses `this.fields` as the iteration base, hence IIUC you mean to add a boolean to `AbstractJsonSchema2Pojo`, right?

Reply:
I mean that we don't need to base this check on the name of the field, but we can set a boolean when mangling the name directly in first place.

Reply:
Right, I am not sure I did what you expected for me to do :smile: ... but I did remove the regex from the hot path and stored the deprecated keys when mangling them. WDYT?

Reply:
it's a good first iteration, agreed 👍